osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -552,9 +577,8 @@ public Task ImportFromStableAsync()
         /// </summary>
         /// <param name="model">The model to populate.</param>
         /// <param name="archive">The archive to use as a reference for population. May be null.</param>
-        protected virtual void Populate(TModel model, [CanBeNull] ArchiveReader archive)
-        {
-        }
+        /// <param name="cancellationToken">An optional cancellation token.</param>
+        protected virtual async Task Populate(TModel model, [CanBeNull] ArchiveReader archive, CancellationToken cancellationToken = default) => await Task.CompletedTask;
```

Comment:
I think this would be a better fit here:
`protected virtual Task Populate(TModel model, [CanBeNull] ArchiveReader archive, CancellationToken cancellationToken = default) => Task.CompletedTask;`

`await` (therefore `async`) is unnecessary, I think overriders will continue to work properly with the surplus `async` keyword. And the reason I am recommending to get rid of async&await is that these keyword pair always generates a new state machine class internally which is completely unnecessary here.

Reply:
Good catch, sloppy async logic.